gems/smithy/lib/smithy/welds/plugins.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Welds
+    # Provides default plugins.
+    class Plugins < Weld
+      def for?(_service)
+        say_status :insert, 'Adding default plugins', @plan.quiet
+        true
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      def plugins
+        {
+          'Smithy::Client::Plugins::Logging' => {
```

Comment:
I'd almost argue that a weld should be responsible for ensuring all of the plugins its exposing are already loaded (ie, it should require all of them) -  I think we could then use the Class instead of a string for the name of it and then I think that simplifies getting the options from it as well.

Reply:
I had this originally but I wasn't sure. I guess that means we don't need to worry about how it gets required when generating options documentation or rbs?

Reply:
Yeah - I think it would simplify that code and I think its a reasonable requirement of a weld